saleor/graphql/discount/tests/deprecated/test_voucher_create.py

```diff
@@ -0,0 +1,271 @@
+import json
+from datetime import timedelta
+from unittest.mock import patch
+
+import graphene
+from django.utils import timezone
+from django.utils.functional import SimpleLazyObject
+from freezegun import freeze_time
+
+from .....core.utils.json_serializer import CustomJsonEncoder
+from .....discount import DiscountValueType, VoucherType
+from .....discount.error_codes import DiscountErrorCode
+from .....discount.models import Voucher
+from .....webhook.event_types import WebhookEventAsyncType
+from .....webhook.payloads import generate_meta, generate_requestor
+from ....tests.utils import get_graphql_content
+from ...enums import DiscountValueTypeEnum, VoucherTypeEnum
+
+CREATE_VOUCHER_MUTATION = """
+mutation  voucherCreate(
+    $type: VoucherTypeEnum, $name: String, $code: String,
+    $discountValueType: DiscountValueTypeEnum, $usageLimit: Int,
+    $minCheckoutItemsQuantity: Int, $startDate: DateTime, $endDate: DateTime,
+    $applyOncePerOrder: Boolean, $applyOncePerCustomer: Boolean) {
+        voucherCreate(input: {
+                name: $name, type: $type, code: $code,
+                discountValueType: $discountValueType,
+                minCheckoutItemsQuantity: $minCheckoutItemsQuantity,
+                startDate: $startDate, endDate: $endDate, usageLimit: $usageLimit
+                applyOncePerOrder: $applyOncePerOrder,
+                applyOncePerCustomer: $applyOncePerCustomer}) {
+            errors {
+                field
+                code
+                message
+            }
+            voucher {
+                type
+                minCheckoutItemsQuantity
+                name
+                code
+                discountValueType
+                startDate
+                endDate
+                applyOncePerOrder
+                applyOncePerCustomer
+            }
+        }
+    }
+"""
+
+
+def test_create_voucher(staff_api_client, permission_manage_discounts):
+    # given
+    start_date = timezone.now() - timedelta(days=365)
+    end_date = timezone.now() + timedelta(days=365)
+    variables = {
+        "name": "test voucher",
+        "type": VoucherTypeEnum.ENTIRE_ORDER.name,
+        "code": "testcode123",
+        "discountValueType": DiscountValueTypeEnum.FIXED.name,
+        "minCheckoutItemsQuantity": 10,
+        "startDate": start_date.isoformat(),
+        "endDate": end_date.isoformat(),
+        "applyOncePerOrder": True,
+        "applyOncePerCustomer": True,
+        "usageLimit": 3,
+    }
+
+    # when
+    response = staff_api_client.post_graphql(
+        CREATE_VOUCHER_MUTATION, variables, permissions=[permission_manage_discounts]
+    )
+    get_graphql_content(response)
+    voucher = Voucher.objects.get()
+
+    # then
+    assert voucher.type == VoucherType.ENTIRE_ORDER
+    assert voucher.name == "test voucher"
+    assert voucher.code == "testcode123"
```

Comment:
It's better to assign such values to variables, in such case when something requires changes, we do not need to change it in two places

Reply:
old tests moved to deprecated folder